src/main/java/com/epam/pages/main/SubjectPage.java

```diff
@@ -0,0 +1,44 @@
+package com.epam.pages.main;
+
+import com.epam.helpers.SharedTestData;
+import com.epam.pages.common.CommonPage;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+
+public class SubjectPage extends CommonPage {
+
+    @FindBy(xpath = "//div[@class='sidebar2']/a[@href='/subjects/Languages/teachers']")
+    private WebElement teachersSection;
+
+    public void clickOnTeachersSection() {
+        uiHelper.clickOnWebElement(teachersSection);
+    }
+
+    public void clickOnAddButton() {
+        uiHelper.clickOnWebElement(createButton);
+    }
+    public int getTeacherListSize() {
+        logger.info("Get teachers list size assigned for subjects");
+        return listItems.size();
+    }
+
+    public void setTeacherListSize() {
+        logger.info("Set teachers list size assigned for subjects");
+        SharedTestData.setListSize(listItems.size());
+    }
+
+    public void clickOnSubjectFromTheList(String subjectName) {
+        logger.info("Click on subject from the list using subject name and index");
+        uiHelper.clickOnWebElement(listItemsHref.get(getIndexOfSubjectFromList(subjectName)));
+    }
+
+    private int getIndexOfSubjectFromList(String subjectName) {
```

Comment:
I think We can try to find a way to solve this method using Java Streams, but in this moment I don't have any idea how to do it. If will find a way to solve it, it will be good.

Reply:
ok